File: hackerrank/codesprint/r1/a5.c
#include<stdio.h>
#include<stdlib.h>

int abs(int a)
{
	if (a<0)
	{
		return -1*a;
	}
	else {
		return a;
	}
}
int calctime(int mouse, int * holes,int n) {
	int time, i;
	time=abs(holes[0]-mouse);
	int newtime;
	int index;
	for (i=0; i<n;i++)
	{
		if (holes[i]!=979762)
		{
			newtime = abs(holes[i]-mouse);
			if (newtime < time) {
				time = newtime;
				index = i;
			}
		}
	}
holes[i]=979762;
return time;
}
int main() {
	int i, n, T;
	int * holes;
	int mouse;
	int timenow, time;
	scanf("%d", &T);
	while (T>0)
	{
	time = -1;
	scanf("%d", &n);
	holes = (int *)calloc(n, sizeof(int));
	for (i=0; i<n; i++)
	{
		scanf("%d", &holes[i]);
	}
	for (i=0; i<n; i++)
	{
		scanf("%d", &mouse);
		timenow = calctime(mouse, holes,n);
		if (time < timenow) {
			time = timenow;
		}
	}
	printf("%d\n", time);

	T=T-1;
	}
return 0;
}

File: hackerrank/codesprint/r1/a5.py
def calcpath(mice, holes):
	time = 0
	pickedhole = 0
	index =0
	for miceloc in mice:
		if miceloc in holes:
			index = holes.index(miceloc)
			del holes[index]
		else:
			least = abs(miceloc - holes[0])
			pickedhole = holes[0]
			for holeloc in holes:
				if least > abs(miceloc - holeloc):
					least = abs(miceloc-holeloc)
					pickedhole = holeloc
			index = holes.index(pickedhole)
			del holes[index]
#		print holes
#		print mice
			if time < least:
				time = least
	print time
def doaction():
	n = raw_input()
	n = int(n)
	newmice = raw_input()
	mice = map(int, newmice.split())
#	mice = [int(each) for each in newmice.split()]
	newholes = raw_input()
#	holes = [int(each) for each in newholes.split()]
	holes = map(int, newholes.split())
	holes.sort()
	mice.sort()
	calcpath(mice, holes)

def init():
	loops=raw_input()
	loops = int(loops)
	while loops>0:
		doaction()
		loops = loops - 1
	exit()
init()
